sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -159,14 +159,22 @@ private static ArraySegment<byte> ReadStreamToArraySegment(Stream stream)
                 {
                     using var memStreamCopy = new MemoryStream((int)(memStreamSource.Length - memStreamSource.Position));
                     memStreamSource.CopyTo(memStreamCopy, StreamBufferSizeInBytes);
```

Comment:
Since the buffer can outlive the stream, do we need to create the copy before reading it, since this case is known to be a memory stream?

Reply:
Fair enough...  nor am I.   I'll ask around internally, but in the absence of information - the safer approach is definitely better.  😄